explainaboard/processors/conditional_generation.py

```diff
@@ -283,26 +300,48 @@ def _gen_cases_and_stats(
         if analysis_level.name == 'example':
             # Note that this is over-ridden to accommodate efficient calculation of
             # EaaS-style metrics
-            inputs = [
-                {
-                    'source': x['source'],
-                    'references': [x['reference']],
-                    'hypothesis': x['hypothesis'],
-                }
-                for x in sys_output
-            ]
-            metric_names = [
-                x.name for x in unwrap_generator(analysis_level.metric_configs)
-            ]
+
+            inputs = []
+            true_data = []
+            pred_data = []
+            for _id, feature_table in enumerate(sys_output):
+                inputs.append(
+                    {
+                        "source": feature_table["source"],
+                        "references": [feature_table["reference"]],
+                        "hypothesis": feature_table["hypothesis"],
+                    }
+                )
+                true_data.append(feature_table["reference"])
+                pred_data.append(feature_table["hypothesis"])
+
+            metric_names_automated = []
+            metric_configs_human = []
+            for metric_config in unwrap_generator(analysis_level.metric_configs):
+                if metric_config.name not in EXTERNAL_METRICS:
+                    metric_names_automated.append(metric_config.name)
+                else:
+                    metric_configs_human.append(metric_config)
```

Comment:
I think the actual difference here is probably not between "automated" and "human" but "EaaS" and "not EaaS". What about making the following change, which would also make the code simpler by removing the dependency on `EXTERNAL_METRICS`. Other changes would be need to made in the following parts of the function (probably just the variable names)

```suggestion
            metric_names_eaas = []
            metric_configs_noneaas = []
            for metric_config in unwrap_generator(analysis_level.metric_configs):
                if isinstance(metric_config, EaaSMetricConfig):
                    metric_names_eaas.append(metric_config.name)
                else:
                    metric_configs_noneaas.append(metric_config)
```

Reply:
good suggestion!